src/tests/ftest/cart/util/cart_logparse.py

```diff
@@ -188,13 +188,13 @@ def __getattr__(self, attr):
                 (filename, _) = self._fields[0].split(':')
                 return filename
             except (IndexError, ValueError):
-                pass
+                return 'Unknown'
```

Comment:
I don't think this is right, and without side-effect.

There are lots of log-lines that don't have a filename, for example the mercury lines which are in the daos file, so there's code which does "if line.hasattr('filename')" which would previously return False but which will now return True with this change.

Reply:
Do you have a suggestion then?   return False here?

or does hasattr catch the exception?

Reply:
Yes, hasattr will catch the exception.

What issue are you trying to resolve here?

Reply:
We are seeing the exception thrown in save_log_line

253         function = getattr(line, 'filename', None)
254         if function:
255             loc = '{}:{}'.format(line.filename, line.lineno)

I suppose we could change this to be more defensive.

I think it happens only when the line otherwise matches expectations but gets truncated.

Reply:
by the way, I don't see any calls to hasattr in cart_log*.py

Reply:
Ok, I think I fixed the root problem here